mne/preprocessing/maxwell.py

```diff
@@ -642,3 +685,116 @@ def _overlap_projector(data_int, data_res, corr):
     return V_principal
+
+
+def _read_fine_cal(info, fine_cal_fname):
+    """Read sensor locations and calib. coeffs from fine calibration file."""
+
+    # Read new sensor locations
+    with open(fine_cal_fname, 'r') as fid:
+        cal_chans = []
+        for line in fid.readlines():
+            if line[0] == '#' or line[0] == '\n':
+                continue
+
+            vals = np.fromstring(line, sep=' ')
+
+            # Check for correct number of items
+            if len(vals) not in [14, 16]:
+                raise RuntimeError('Error reading fine calibration file')
+
+            ch = int(vals[0].copy())  # Get channel number
+            ch_name = 'MEG' + '%04d' % ch  # Zero-pad names to 4 characters
+            loc = vals[1:13].copy()  # Get orientation information
+            calib_coeff = vals[13:].copy()  # Get imbalance/calibration coeff
+            ch_type = channel_type(info, info['ch_names'].index(ch_name))
+
+            cal_chans.append(dict(ch=ch, ch_name=ch_name, loc=loc,
+                                  calib_coeff=calib_coeff, ch_type=ch_type))
+
+    # Check that we ended up with correct number of channels
+    if len(cal_chans) != info['nchan']:
+        raise RuntimeError('Number of channels in fine calibration file (%i) '
+                           'does not equal number of channels in info (%i)' %
+                           (len(cal_chans), info['nchan']))
+
+    return cal_chans
+
+
+def _update_sensor_geometry(info, cal_chans):
+    """Helper to replace sensor geometry information"""
+
+    # Replace sensor locations (and track differences) for fine calibration
+    ang_shift = np.zeros((info['nchan'], 3))
+    for cal_chan in cal_chans:
+        ch_ind = info['ch_names'].index(cal_chan['ch_name'])
+
+        # Loop over vectors in X, Y, and Z directions
+        for ang_i, (i1, i2) in enumerate(zip(range(3, 10, 3),
+                                             range(6, 13, 3))):
+            v1 = cal_chan['loc'][i1:i2]
+            v2 = info['chs'][ch_ind]['loc'][i1:i2].astype(float)
+
+            cos_ang = np.dot(v1, v2) / (np.linalg.norm(v1) *
+                                        np.linalg.norm(v2))
+            ang_shift[ch_ind, ang_i] = cos_ang
+
+        # Adjust channel orientation with those from fine calibration
+        info['chs'][ch_ind]['loc'] = cal_chan['loc']
+
+    # Deal with numerical precision giving values slightly more than 1.
+    np.clip(ang_shift, -1., 1., ang_shift)
+    ang_shift = np.rad2deg(np.arccos(ang_shift))  # Convert to degrees
+
+    # Log quantification of sensor changes
+    logger.info('Fine calibration adjusted coil positions by (mean ± std in '
+                'degrees): %0.1f ± %0.1f (max: %0.1f)' %
+                (np.mean(ang_shift), np.std(ang_shift),
+                 np.max(np.abs(ang_shift))))
+
+    return info
+
+
+def _sss_basis_point(origin, info, int_order, ext_order, imbalance,
+                     head_frame=True):
```

Comment:
I added it for debugging purposes because it was a little easier for Samu to give me matrices for comparison in the device coordinate frame. I can change it back though